File: Data Structures & Algorithms/pow-x-n/submission-1.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if(n == 0)return 1;
        double num= x;
        bool check = false ;
        if(n<0){
            check = true;
            n = n * (-1);
        }
        for(int i =1;i<n;i++){
            x = num*x;
        }
        if(check == true){
            return (double)1/x;
        }
        return x;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-1.java
class Solution {
    public boolean isHappy(int n) {
                Map<Integer,Integer>map = new HashMap<>();
        while(n!= 1){
            int sum = 0;
            int num = n;
            while(num!= 0){
                sum += (int) Math.pow(num % 10, 2);
                num = num/10;
            }

            if(map.containsKey(sum)){
                return false;
            }

            map.put(sum,1);
            n = sum;

        }

        return true;
    }
}
